pex/bin/pex.py

```diff
@@ -826,11 +836,14 @@ def to_python_interpreter(full_path_or_basename):
         pex_builder.add_from_requirements_pex(requirements_pex)
 
     with TRACER.timed("Resolving distributions ({})".format(reqs + options.requirement_files)):
+        if options.cache_ttl:
+            pex_warnings.warn("The --cache-ttl option is deprecated and no longer has any effect.")
```

Comment:
Worth referencing on https://github.com/pantsbuild/pex/issues/1084 I suspect.

Rather than fine-tuning TTLs moving forward, I expect that true lockfile support (#1086 - which would include the hashes/URLs to fetch, rather than actually running a resolve) would be the best optimization.

Reply:
The TTLs were actually an anti-optimization in practice (except for Twitter's hack to reduce load on find-links repos). Any positive TTL causes Pip to ignore its cache once the TTL expires whereas a 0 TTL causes it to use the cache with a conditional GET as long as the resource stays the same. So this appears to be a strict win except for the Twitter case. I've updated #1084 with some comments about this upgrade. And agreed, #1086 will be the solution Pants was looking for although fwict it will involve working outside Pip. Command line tests show that Pip still hits the network with --no-deps and --hash-checking enabled. You get a fixed resolve, but its slower than need be.